cfgov/jinja2/v1/_includes/macros/video-player.html

```diff
@@ -27,27 +27,32 @@
 
 {% macro render(options={}) -%}
     {% import 'macros/share.html' as share with context %}
-    {% set video_player = options.video.player | default('youtube') %}
-    {% set video_id     = options.video.id     | default('') %}
-    {% set video_url    = options.video.url    | default('/') %}
-    {% set image_url    = options.image.url or
-                          'https://img.youtube.com/vi/' + video_id + '/maxresdefault.jpg'
-                          if video_player == 'youtube'
-    %}
+    {% set video_url    = options.video.url | default('/') %}
+    {% set video_player = options.video.player or
+                          ( video_url.find( 'youtube.com') > -1 and 'youtube' ) or
+                          ( video_url.find( 'ustream.tv') and 'ustream' ) %}
+    {#  TODO: Replace with django method for getting video id if youtube #}
+    {% if video_player == 'youtube' %}
+        {% set video_id = options.video.id or ( video_url.split('?')[0].split('/') | last ) %}
+        {% set image_url = options.image.url or ( 'https://img.youtube.com/vi/' + video_id + '/maxresdefault.jpg' ) %}
+    {% else %}
+        {% set video_id = '' %}
+        {% set image_url = '/static/img/cfpb_video_cover_card_1380x776.png' %}
```

Comment:
This should also have the option for the CMS delivered image `options.image.url`